internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -33,6 +30,13 @@ type CloudFormation struct {
 	box    packd.Box
 }
 
+type stackConfiguration interface {
+	StackName() string
+	Template() (string, error)
+	Parameters() []*cloudformation.Parameter
+	Tags() []*cloudformation.Tag
+}
```

Comment:
Just thinking out loud here - it seems like this should be part of the public facing API? If DeployEnvironment (as an example) took in a `StackConfiguration`, we could potentially decouple the package further from "environment deploys"?

Reply:
It will be dope to have a generic CF templating "engine". But would it be more efficient to do that in another PR so that you are unblocked? (thought you are blocked on this)

Reply:
To further expand on this it looks like a lot of the internal logic could potentially collapse and simplify as I see we're consistently creating a `stackConfiguration` from an `archer.DeployEnvironmentInput`. If `archer.DeployEnvironmentInput` implemented `stackConfiguration` we could pass that in and rely on the `stackConfiguration` interface abstraction to do CloudFormation things.

It could be worth investigating? Not necessarily part of this PR.

Reply:
I flirted with doing something like that, actually! But I went this route to keep the deploy interface ignorant of CloudFormation - as much as possible. This stackConfiguration is essentially methods on top of that input to transform it into CF stuff.

I'm thinking in the future, if we support another deployment target, this might be a helpful abstraction (and it doesn't cost too much).

Reply:
Nice! Didn't think of that (i.e. support non-CF deployment). To echo Nick's suggestion, since we have multiple places in the code that dynamically generate CF templates, I suppose there's still value to create some form of CF templating "engine". Will know more for sure once the work for Archer pipeline CFN template starts.